test/jira2md.js

```diff
@@ -1,128 +1,162 @@
-        var markdown = j2m.to_markdown("# Foo\n# Bar\n# Baz\n## FooBar\n## BarBaz\n### FooBarBaz\n# Starting Over");
-        markdown.should.eql("1. Foo\n1. Bar\n1. Baz\n  1. FooBar\n  1. BarBaz\n    1. FooBarBaz\n1. Starting Over");
-    });
-    it('should handle bold AND italic (combined) correctly', function() {
-        var markdown = j2m.to_markdown("This is _*emphatically bold*_!");
-        markdown.should.eql("This is ***emphatically bold***!");
-    });
-    it('should handle bold within a un-ordered list item', function() {
-        var markdown = j2m.to_markdown("* This is not bold!\n** This is *bold*.");
-        markdown.should.eql("* This is not bold!\n  * This is **bold**.");
-    });
-    it('should be able to handle a complicated multi-line jira-wiki string and convert it to markdown', function() {
-        var jira_str = fs.readFileSync(path.resolve(__dirname, 'test.jira'),"utf8");
-        var md_str = fs.readFileSync(path.resolve(__dirname, 'test.md'),"utf8");
-        var markdown = j2m.to_markdown(jira_str);
-        markdown.should.eql(md_str);
-    });
-    it('should not recognize strikethroughs over multiple lines', function() {
-        var markdown = j2m.to_markdown("* Here's an un-ordered list line\n* Multi-line strikethroughs shouldn't work.");
-        markdown.should.eql("* Here's an un-ordered list line\n* Multi-line strikethroughs shouldn't work.");
-    });
-    it('should remove color attributes', function() {
-        var markdown = j2m.to_markdown("A text with{color:blue} blue \n lines {color} is not necessary.");
-        markdown.should.eql("A text with blue \n lines  is not necessary.");
-    });
+describe('to_markdown', function () {
+  it('should exist', function () {
+    should.exist(j2m.to_markdown);
+  });
+
+  it('should be a function', function () {
+    j2m.to_markdown.should.be.a('function');
+  });
+
+  it('should convert bolds properly', function () {
+    var markdown = j2m.to_markdown('*bold words*');
+    markdown.should.eql('**bold words**');
+  });
+
+  it("does not perform intraword formatting on asterisks", function() {
+    var markdown = j2m.to_markdown("a*phrase*with*asterisks");
+    markdown.should.eql("a*phrase*with*asterisks");
+  });
+
+  it('should convert italics properly', function () {
+    var markdown = j2m.to_markdown('_italic_');
+    markdown.should.eql('*italic*');
+  });
+
+  it('does not perform intraword formatting on underscores', function () {
+    var markdown = j2m.to_markdown('a_phrase_with_underscores');
+    markdown.should.eql('a_phrase_with_underscores');
+  });
+
+  it('should convert monospaced content properly', function () {
+    var markdown = j2m.to_markdown('{{monospaced}}');
+    markdown.should.eql('`monospaced`');
+  });
+
+  //it('should convert citations properly', function() {
+  //    var markdown = j2m.to_markdown('??citation??');
+  //    markdown.should.eql('<cite>citation</cite>');
+  //});

```

Comment:
What happened to the citations?

Reply:
Excellent question. Looks like it's been broken for a while? I'll add it to my list.

Reply:
Hmm. So @taoza the same question was asked on [this PR](https://github.com/kylefarris/J2M/pull/4) where the issue with citations is described as:

> some infinite loops it was causing in certain circumstances 

I'm wary of adding the code back without understanding what those circumstances might be.